qlib/kernel/loader/vdso.rs

```diff
@@ -78,6 +90,34 @@ impl VdsoInternal {
             }
         }
 
+        //Symbol table section
+        let symtab_section = elfFile.find_section_by_name(".symtab")
+                                    .ok_or("VDSO: no .symtab section.");
+        if symtab_section.is_ok() {
+            let symtab_data = symtab_section.unwrap().get_data(&elfFile);
+            use xmas_elf::sections::SectionData::SymbolTable64;
+            use xmas_elf::symbol_table::Entry;
+            if let Ok(SymbolTable64(symbols)) = symtab_data {
+                let mut vdso_symtab:VdsoSymbols = Default::default();
+                error!("VDSO: parse symbol table, len:{}.", symbols.len());
+                for entry in 0..symbols.len() {
+                    info!("VDSO: symbol {} ", symbols[entry].get_name(&elfFile).unwrap());
+                    match symbols[entry].get_name(&elfFile) {
+                        Ok("__kernel_clocl_gettime") =>
+                            vdso_symtab.KernelClockGetTime = symbols[entry].value(),
+                        Ok("__kernel_gettimeofday") =>
+                            vdso_symtab.KernelGetTimeOfDay = symbols[entry].value(),
+                        Ok("__kernel_clock_getres") =>
+                            vdso_symtab.KernelClockGetRes = symbols[entry].value(),
+                        Ok("__kernel_rt_sigreturn") =>
+                            vdso_symtab.KernelRtSigreturn = symbols[entry].value(),
+                        _ => {},
+                    }
+                }
+                self.vdso_symbols = Some(vdso_symtab);
```

Comment:
What if the `vdso_symtab` is empty after the loop? You would end up with `Some(VdsoSymbols{0,0,0,0})`, this is semantically `None`

Reply:
My thought strategy is to add a 'check and panic', at least for Arm as it depends on one of the symbols for signals to work. I doubt there is a recovery strategy...
But for now I was only trying to retrieve the symbols.